rclcpp_action/src/client_base.cpp

```diff
@@ -399,7 +399,7 @@ ClientBase::handle_goal_response(
 void
 ClientBase::send_goal_request(std::shared_ptr<void> request, ResponseCallback callback)
 {
-  std::unique_lock<std::mutex> guard(pimpl_->goal_requests_mutex);
+  std::lock_guard<std::recursive_mutex> guard(pimpl_->goal_requests_mutex);
```

Comment:
i do not find the reason why it needs to be `std::unique_lock` here.

Reply:
should make no difference, this constructor of unique_lock would lock on construction and unlock on destruction.